src/services/exit_order.py

```diff
@@ -1,24 +1,285 @@
+    ) -> dict[NodeOperatorIndex, int]:
+        """
+        Get last requested to exit validator index for each operator
+        """
+        module_operator_ids = {}
+        for module_id, operator_id in operator_validators:
+            module_operator_ids.setdefault(module_id, set()).add(operator_id)
+
+        last_requested_to_exit_indexes = {}
+        for module_id, operator_ids in module_operator_ids.items():
+            operator_ids.sort()
+            per_operator_indexes = self._get_last_requested_validator_index(blockstamp, module_id, operator_ids)
+            for operator_id in operator_ids:
+                last_requested_to_exit_indexes[(module_id, operator_id)] = per_operator_indexes[operator_id]
+        return last_requested_to_exit_indexes
+
+    def _get_delayed_validators_per_operator(
+        self,
+        blockstamp: BlockStamp,
+        operator_validators: dict[NodeOperatorIndex, list[LidoValidator]],
+        last_requested_to_exit_indices_per_operator: dict[NodeOperatorIndex, int],
+    ) -> dict[NodeOperatorIndex, int]:
+        """
+        Get delayed validators count for each operator
+        """
+
+        delayed_validators_per_operator = defaultdict(int)
+
+        # We don't consider validator as delayed if it was requested to exit
+        # in last VALIDATOR_DELAYED_TIMEOUT_IN_SLOTS slots
+        recently_requested_to_exit_indices_per_operator = self._get_recently_requested_to_exit_indices(
+            blockstamp, operator_validators
+        )
+
+        for module_operator, validators in operator_validators.items():
+            recently_requested_to_exit_indices = recently_requested_to_exit_indices_per_operator[module_operator]
+            for validator in validators:
+                previously_requested_to_exit = (
+                    int(validator.index) <= last_requested_to_exit_indices_per_operator[module_operator]
+                )
+                on_exit = self._is_on_exit(validator)
+                recently_requested_to_exit = int(validator.index) in recently_requested_to_exit_indices
+                if previously_requested_to_exit and not on_exit and not recently_requested_to_exit:
+                    delayed_validators_per_operator[module_operator] += 1
+
+        return delayed_validators_per_operator
+
+    def _get_recently_requested_to_exit_indices(
+        self, blockstamp: BlockStamp, operator_validators: dict[NodeOperatorIndex, list[LidoValidator]]
+    ) -> dict[NodeOperatorIndex, set[int]]:
+        """
+        Returns recently requested to exit validators indices per operator
+        """
+        module_operator = {}
+
+        # Initialize dict with empty sets for operators which validators were not contained in any event
+        for operator in operator_validators:
+            module_operator[operator] = set()
+
+        from_block = blockstamp.block_number - VALIDATOR_DELAYED_TIMEOUT_IN_SLOTS
```

Comment:
to be sure that we are in the valid range

```suggestion
        from_block = max(0, blockstamp.block_number - VALIDATOR_DELAYED_TIMEOUT_IN_SLOTS)
```

Reply:
fixed